pkg/controller/mongodb/replicaset_controller_test.go

```diff
@@ -301,6 +303,35 @@ func TestAutomationConfig_versionIsNotBumpedWithNoChanges(t *testing.T) {
 	assert.Equal(t, currentAc.Version, 1)
 }
 
+func TestAutomationConfig_CustomMongodConfig(t *testing.T) {
+	mdb := newTestReplicaSet()
+
+	mongodConfig := objx.New(map[string]interface{}{})
+	mongodConfig.Set("net.port", 1000)
+	mongodConfig.Set("arbitrary.config.path", "value")
```

Comment:
this is a very nice interface for accessing nested values!

Reply:
Agreed!